### PR TITLE
Allow custom timestamp from user input

### LINKED ISSUE #4: Timestamp decoding should be extendible from the command line
Adding a new timestamp format should be as easy as providing a regex which matches the format and a `strptime` expression that decodes it to a `datetime`.

### PR DESCRIPTION
**Pull request by pro-panda**
Fixes #4 

Example
```
python3 logmerge.py file1 file2 -r='(\d\d\d\d-\d\d-\d\d \d\d:\d\d:\d\d\.\d\d\d) ' -f='%Y-%m-%d %H:%M:%S.%f'
```

### CODE REVIEW

File: logmerge.py [hunk 1]
```diff
@@ -169,6 +177,16 @@ def main():
     if args.logfiles is None or len(args.logfiles) < 2:
         print("Requires at least two logfiles")
         exit(1)
+    elif args.format is None and args.regex is not None:
+        print("Requires timestamp regex along with format")
+        exit(1)
+    elif args.regex is None and args.format is not None:
```
[the remark this answers is not in the document]
Comment (pro-panda) [reply logmerge.py]: Thanks. 
I made separate clauses for different error messages. I'll switch to a common message

**Comment by pro-panda** [comment]
Changes made.

**Comment by jasonzio** [comment]
Looks good to me.

**Comment by pro-panda** [comment]
This PR is faulty. See #10 for fix

**Pull request merged**